### ISSUE TITLE
Audio Book Files Listing Randomly

### ISSUE THREAD

**Issue by TomAlperin** [ejurgensen/forked-daapd#91]
Sound files that are added to the library as audio books do not properly have their "chapter" files sorted and are instead listed in what appears to be a random order. Songs that are in a music album however are sorted properly by their track number.

**Comment by TomAlperin** [comment]
This appeared to have been fixed on previous versions, however is now happening again. Audio book collections that were previously added to the library are still in the proper order, however new audio book collections are not listing properly. I think it might be possible that the partial rescan does not add audiobook to the library in the proper order, however a full rescan may do this correctly. Will experiment with this and report back.

**Comment by ejurgensen** [comment]
I don't think this was ever fixed, maybe you just got lucky with the ordering for a while. @chme has submitted a fix for this, but I will admit I have been a bit slow to follow up on it. I would like to first see how iTunes deals with this. If you can point me to a free/cheap multi-chapter/file Audiobook in the iTunes Store it would be very helpful for testing.

**Comment by TomAlperin** [comment]
My "Audiobook" collections are actually old time radio. I've gotten many of them from the Internet Archive, which is free, however most of them fail to put track numbers in, requiring me to do this myself in iTunes. They are also not marked as Audiobooks and I make this change in iTunes as well before adding them to forked-daapd. The last one I added was called "Speed Gibson of the International Secret Police." This would actually be the only one, I think that didn't require I add the track numbers. One thing that I did notice is that when I selected to play the first track without shuffle, they tracks were added to the cue in the correct order in spite of them being listed out of order under Audiobooks. Not really predictable behavior, but at least I am getting the end result that I would prefer in the end.
